pkg-builder.pl

```diff
@@ -350,6 +350,7 @@ ()
    cpy_file( "store/conf/attrs/amavisd-new-attrs.xml", "$stage_base_dir/opt/zimbra/conf/attrs/amavisd-new-attrs.xml" );
    cpy_file( "store/conf/attrs/zimbra-attrs.xml",      "$stage_base_dir/opt/zimbra/conf/attrs/zimbra-attrs.xml" );
    cpy_file( "store/conf/attrs/zimbra-ocs.xml",        "$stage_base_dir/opt/zimbra/conf/attrs/zimbra-ocs.xml" );
+   cpy_file( "store/conf/attrs/zimbra-attrs-schema",        "$stage_base_dir/opt/zimbra/conf/attrs/zimbra-attrs-schema" );
```

Comment:
Note: This change causes the following to appear:

`
zimbra@zcs-dev-two:~$ zmprov gcf zimbraLDAPSchemaVersion

[] WARN: while loading attrs, ignoring not .xml file: /opt/zimbra/conf/attrs/zimbra-attrs-schema
zimbraLDAPSchemaVersion: 1520374873`

Reply:
Hmm.  That suggests the attrs/ sub-directory probably isn't a good location for the file.  Either that or some clever stuff is needed to suppress the error.

Reply:
I was afraid that would be the answer.  Any other suggestions? 
I thought maybe a hidden file would work:

`[] WARN: while loading attrs, ignoring not .xml file: /opt/zimbra/conf/attrs/.zimbra-attrs-schema`

Alas it was not to be.

Reply:
How about in /opt/zimbra/conf?

Reply:
Offending code in AttributeManager.java:288

       for (File file : files) {
            if (!file.getPath().endsWith(".xml")) {
                ZimbraLog.misc.warn("while loading attrs, ignoring not .xml file: %s", file);
                continue;
            }
            if (!file.isFile()) {
                ZimbraLog.misc.warn("while loading attrs, ignored non-file: %s", file);
            }

Any reason we couldn't just drop those lines entirely? If the code doesn't care about anything but XML files is there any reason to warn someone and ignore the file when it could just be ignored with no warning.

Reply:
Looks like we have some choices now:

- move the file to /opt/zimbra/conf
- update xml finder code to ignore non-xml files silently
- rename the file to `.zimbra-attrs-schema` and update code to ignore files with a leading '.' character

Reply:
FYI: Going with option 1: move the file to /opt/zimbra/conf